fog/view/server/src/db_fetcher.rs

```diff
@@ -282,64 +291,67 @@ impl<DB: RecoveryDb + Clone + Send + Sync + 'static> DbFetcherThread<DB> {
             // Attempt to load data for the next block.
             let get_tx_outs_by_block_result = {
                 let _metrics_timer = counters::GET_TX_OUTS_BY_BLOCK_TIME.start_timer();
-                self.db
-                    .get_tx_outs_by_block_and_key(ingress_key, block_index)
+                self.db.get_tx_outs_by_block_range_and_key(
+                    ingress_key,
+                    block_index,
+                    self.block_batch_request_size as u64,
+                )
             };
 
             match get_tx_outs_by_block_result {
-                Ok(Some(tx_outs)) => {
-                    let num_tx_outs = tx_outs.len();
-
-                    // Log
-                    log::info!(
-                        self.logger,
-                        "ingress_key {:?} fetched {} tx outs for block {}",
-                        ingress_key,
-                        num_tx_outs,
-                        block_index,
-                    );
-
-                    // Ingest has produced data for this block, we'd like to keep trying the
-                    // next block on the next loop iteration.
-                    may_have_more_work = true;
-
-                    // Mark that we are done fetching data for this block.
-                    self.block_tracker.block_processed(ingress_key, block_index);
-
-                    // Store the fetched records so that they could be consumed by the enclave
-                    // when its ready.
-                    {
-                        let mut state = self.shared_state();
-                        state.fetched_records.push(FetchedRecords {
+                Ok(block_results) => {
+                    if !block_results.is_empty() {
```

Comment:
Optional/Equivalent: Negate this condition to unindent the sizable block below
```suggestion
                    if block_results.is_empty() continue;
```